### DISCUSSION TITLE
AppSync doesn't support blockstring descriptors

### DISCUSSION THREAD

**Discussion by genehynson** [maoosi/prisma-appsync#151] [Q&A] [answered]
Hey there! I'm trying to deploy my prisma-appsync project to AppSync but it's failing with `Schema Creation Status is FAILED with details: Failed to parse schema document - ensure it's a valid SDL-formatted document.`

I'm pretty sure this is because of the blockstring descriptors in the schema file - AWS [AppSync notoriously doesn't support this standard](https://github.com/aws/aws-appsync-community/issues/38).

Is there a way to generate the schema without these blockstring descriptors to get around this issue? Thanks!

e.g. this causes issues:

```
        """
        Find a single User record by unique identifier.
        """
        getUser(
            where: UserWhereUniqueInput!
            ): User 
```

**Comment by maoosi** [comment]
Hey @genehynson!

If you are using AWS CDK to deploy your API, then BlockString Descriptions are automatically removed (not doing anything on our end, this is something the AWS CDK lib is doing).

Unless you are trying to copy/past your schema directly into the AppSync console UI, then I would think the issue lies somewhere else. Would you be able to share your Prisma Schema file?

**Reply by genehynson** [reply]
Hey @maoosi ! Thanks for that information! Yes, I was pasting the schema into the AppSync console UI assuming the same validation occurs - good to know that's not the case!

I used [@graphql-validate/cli](https://www.npmjs.com/package/@graphql-validate/cli) on the schema and it errored saying with the following message: `GraphQLError: Syntax Error: Expected Name, found "}".` And pointed to this block in my schema:

```
    input UserActivityWhereUniqueInput {
        
    }
```

It does seem like this generated fragment is missing some content? Any ideas? Here's my [schema file](https://drive.google.com/file/d/1asHgNclVZQKp_DR7Gp6rmUPdi8sHqU9u/view?usp=sharing) (google drive link - it's rather large)

Thanks!

**Reply by maoosi** [reply, marked as answer]
Hey @genehynson! This looks like an issue on Prisma-AppSync's end. Would you mind opening an issue and including the portion of your Prisma Schema (`schema.prisma` file) related to your `User` and `UserActivity` models? This would help us understand why an empty block is being generated.
